test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,185 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+// written for the purpose of the Bch Softwarepraktikum, early 2019 FU Berlin
+// @author: Clemens Cords <clemenscords@fu-berlin.de>
+
+#include <fstream>
+#include <cstring>
+#include <cmath>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/io/sequence_file/format_fastq.hpp>
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/quality/phred42.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/range/view/to_char.hpp>
+
+using namespace seqan3;
+
+const std::size_t POWER = 5; //files will have |10^POWER| lines
+
+const auto DNA_SEQ = "AGCTAGCAGCGATCGCGATCGATCAGCGATCGAGGAATATAT"_dna4;
+const auto QUALITY = "IIIIIHIIIIIIIIIIIIIIIIIIIIIIHHGIIIIHHGIIIH"_phred42;
+
+// ============================================================================
+// generate dummy fastq file with 10^power identical 4-line entries
+// ============================================================================
+std::string generate_dummy_fastq_file(std::size_t power)
+{
+    std::string file{};
+
+    const std::string id{"@name"};
+    const std::string seq = DNA_SEQ | view::to_char;
+    const std::string qualid{"+"};
+    const std::string quality = QUALITY | view::to_char;
+
+    const unsigned int n_iterations = static_cast<unsigned int>(pow(10, power));
+
+    for (size_t i = 0; i < n_iterations; ++i)
+    {
+
+        file += id + '\n'
```

Comment:
Does this fit into one line (180 char)? 
Also putting "+" in an extra string is a little too generic for the scope of a benchmark that is not exposed to users. you can directly insert the characters here.

EDIT: you seem to use them a lot in the write benchmarks too, so you might just add them once as a global variable and use them throughout the benchmarks.